src/sources/prometheus/scrape.rs

```diff
@@ -90,6 +93,37 @@ impl SourceConfig for PrometheusScrapeConfig {
             .endpoints
             .iter()
             .map(|s| s.parse::<http::Uri>().context(sources::UriParseSnafu))
+            .map(|r| {
+                if let Ok(uri) = r {
+                    let mut serializer = url::form_urlencoded::Serializer::new(String::new());
+                    if let Some(query) = uri.query() {
+                        for (k, v) in url::form_urlencoded::parse(query.as_bytes()) {
+                            serializer.append_pair(&k, &v);
+                        }
+                    };
+                    if let Some(query) = &self.query {
+                        for (k, l) in query {
+                            for v in l {
+                                serializer.append_pair(&k, &v);
+                            }
+                        }
```

Comment:
I have a big feeling that this can be written much better!
Same for the for-loop slightly above.

Reply:
Will review the CI results on this code. Probably the compiler is right that I shouldn't have used a dereference.